specification/programmableconnectivity/Azure.ProgrammableConnectivity/apis/number.tsp

```diff
@@ -0,0 +1,92 @@
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace Azure.ProgrammableConnectivity;
+
+/// Interfaces
+
+@doc("""
+  Number operations include Frontend Authentication.
+  
+  This means that each operation has 2 responses. 
+  
+  The first response is a 302 redirect, which redirects the device to the device's Network
+  to authenticate directly. The Network responds with another 302 redirect and a token.
+  This redirects the device to APC, where the token is used to verify / retrieve
+  the device number. The second response is a 200 containing the result of the query.
+
+  This flow takes place automatically without user interaction, and so only the result
+  of the entire flow (the 200 response) is included in this document.
+""")
+interface NumberInterface {
+  @doc("Verifies the phone number (MSISDN) associated with a device.")
+  verify is Operations.ResourceAction<
+    NumberEndpoint,
+    NumberVerifyRequest,
+    NumberVerifyResponse
+  >;
+
+  @doc("Retrieves the phone number (MSISDN) associated with a device.")
+  retrieve is Operations.ResourceAction<
+    NumberEndpoint,
+    NetworkIdentifier,
+    NumberRetrieveResponse
+  >;
+}
+
+/// Endpoints
+
+@resource("Apis")
```

Comment:
**Thought:** It seems weird to have four different Endpoint objects with the decorator @resource("Apis"). On the other hand I think this is how we have to do it - if we combine all the Endpoints into one, it's not obvious to me how make it clear which path maps to which API. Do you have any more thoughts about this?

Reply:
You're right, I think its ugly, and its definitely more ugly since setting this up as a proper enum. I've had a little play around with trying to combing the models, but without much success- have got a few things to compile but they don't look like what we want. I'll potentially ask in the typespec channel, or just ask explicitly if we could be doing something better in the API review meeting